test/fuzz/json/json.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO TODO
+// Package json implements fuzzers for json deserialization routines in
+// Kubernetes. These targets are compatible with the github.com/dvyukov/go-fuzz
+// fuzzing framework.
+package json
+
+import (
+	"bytes"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func FuzzStrictDecode(data []byte) int {
+	gvk := &schema.GroupVersionKind{Version: "v1"}
+	strictOpt := SerializerOptions{false, false, true}
+	strictYamlOpt := SerializerOptions{true, false, true}
+	strictPrettyOpt := SerializerOptions{false, true, true}
+	nonstrictOpt := SerializerOptions{false, false, true}
+	scheme := runtime.NewScheme()
+	strictSer := NewSerializerWithOptions(DefaultMetaFactory, scheme, scheme, strictOpt)
+	nonstrictSer := NewSerializerWithOptions(DefaultMetaFactory, scheme, scheme, nonstrictOpt)
+	ysSer := NewSerializerWithOptions(DefaultMetaFactory, scheme, scheme, strictYamlOpt)
```

Comment:
Can we move yaml fuzz tests into test/yaml so we can use different corpora?  Maybe this can be a TODO.

Reply:
Acutally, since yaml is a superset of json, we probably don't want to change these.